packages/experimental/AppearanceAdditions/ios/FRNAppearanceAdditions.m

```diff
@@ -97,31 +97,27 @@ + (BOOL)requiresMainQueueSetup {
     return @[ @"appearanceChanged" ];
 }
 
-- (dispatch_queue_t)methodQueue
-{
+- (dispatch_queue_t)methodQueue {
     return dispatch_get_main_queue();
 }
 
 - (void)startObserving {
     _hasListeners = YES;
     
-    // Note that [UITraitCollection currentTraitCollection] always returns the same default trait collection,
-    // presumably because FRNAppearanceAdditions isn't a view, so it never gets updated with the right traitCollection
-    // (which happens when a view gets added to the view hierachy). In order to get the right trait collection,
-    // we need to access a view that's been added to the view hierarchy
-    UIViewController *viewControllerWithInitialTraitCollection = RCTPresentedViewController();
-    UITraitCollection *initialTraitCollection;
-    
-    if (viewControllerWithInitialTraitCollection != nil) {
-        initialTraitCollection = [viewControllerWithInitialTraitCollection traitCollection];
-    } else {
-        initialTraitCollection = [UITraitCollection currentTraitCollection];
+    for (UIWindow *window in RCTSharedApplication().windows) {
+        RCTRootView *rootView = (RCTRootView *)[[window rootViewController] view];
```

Comment:
I also mentioned this on Discord: With hybrid apps, I don't think this is always true. `RCTRootView` can live further down the tree.

Getting the node handle should give you a more accurate result. The issue with this approach is that it may require thread jumps. You need to resolve the tag inside `RCTExecuteOnMainQueue` to safely access `RCTUIManager`.

Reply:
Thanks for pointing that out timmy - we also realized that trying to find a RCTRootView from UISharedApplication wasn't the right approach since RCTRootView might not be the actual root view. I've removed this code & now this PR relied on passing the viewTag from the JS side to get the right trait collection. 

I was able to use `RCTExecuteOnMainQueue` to get this working - thanks for letting me know about that - and ended up not needing to get the node handle